plugins/blockupdates/main.go

```diff
@@ -198,20 +199,46 @@ func (bu *blockUpdatesModule) InitializeNode(stack *node.Node, b types.Backend)
 	cache, _ = lru.New(128)
 	recentEmits, _ = lru.New(128)
 	suCh = make(chan *stateUpdateWithRoot, 128)
+
+	hasLast, err := b.ChainDb().Has([]byte("lastPrunedStateUpdate"))
+	if err != nil {
+		log.Error("error returned checking for last pruned block", "err", err)
+	}
+
+	if !hasLast {
+		currentBlock := b.CurrentBlock()
+		lastPruned = currentBlock.Number.Uint64()
```

Comment:
I think we need to write lastPruned to the database here. I don't think it will happen below unless the server is running long enough to actually prune a block, and at 45k blocks any server that shuts down in less than ~7 days isn't going to actually prune any blocks, but we still want the pruner to know where to pick up on the next startup.